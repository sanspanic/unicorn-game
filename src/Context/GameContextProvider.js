import React, { useState } from "react";
import GameContext from "./GameContext";

const GameContextProvider = ({ children }) => {
  const initialState = {
    status: "won",
    difficulty: 0,
    size: 15,
    name: "",
  };
  const [gameData, setGameData] = useState(initialState);

  //restarts game by setting gameData.status back to "setup" & redirecting to homepage
  const playAgain = () => {
    setGameData({ ...gameData, status: "setup" });
  };

  return (
    <GameContext.Provider value={{ gameData, setGameData, playAgain }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContextProvider;
