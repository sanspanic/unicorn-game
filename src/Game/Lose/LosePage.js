import React, { useContext } from "react";
import GameContext from "../../Context/GameContext";

const LosePage = () => {
  const { playAgain } = useContext(GameContext);
  return (
    <div>
      YA LOST
      <button onClick={playAgain}>Play again! </button>
    </div>
  );
};

export default LosePage;
