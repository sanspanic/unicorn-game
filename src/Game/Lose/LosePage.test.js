import React from "react";
import { fireEvent, render } from "@testing-library/react";
import LosePage from "./LosePage";
import GameContextProvider from "../../Context/GameContextProvider";
import { MemoryRouter } from "react-router-dom";

test("renders without crashing", () => {
  render(
    <GameContextProvider>
      <LosePage />
    </GameContextProvider>
  );
});

test("matches snapshot", () => {
  const { asFragment } = render(
    <GameContextProvider>
      <LosePage />
    </GameContextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
