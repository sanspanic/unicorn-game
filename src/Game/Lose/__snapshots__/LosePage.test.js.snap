// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-evenly items-center p-10 h-screen"
  >
    <div
      class="font-mono text-xl text-center"
    >
      You lost! :(
    </div>
    <div
      class="react-floater-animated"
    >
      <img
        alt="happy floating skull monster"
        class="mx-auto w-10/12 sm:w-9/12 md:w-5/12 lg:w-3/12"
        src="Skull.png"
      />
    </div>
    <button
      class="shadow-xl py-2 px-3 border-2 border-purple-700 rounded gradient-pink text-white font-mono"
    >
      Restart
    </button>
  </div>
</DocumentFragment>
`;
